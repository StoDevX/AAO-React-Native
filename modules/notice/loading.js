// @flow
import * as React from 'react'
import {NoticeView} from './notice'

export function LoadingView({text = 'Loadingâ€¦'}: {text?: string}) {
	return <NoticeView spinner={true} text={text} />
}
