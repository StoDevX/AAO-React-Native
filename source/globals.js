// @flow

export const GH_PAGES_URL = (relativePath: string) => {
	return `https://stodevx.github.io/AAO-React-Native/${relativePath}`
}

export const GH_NEW_ISSUE_URL =
	'https://github.com/StoDevX/AAO-React-Native/issues/new'
