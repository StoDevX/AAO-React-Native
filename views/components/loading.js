// @flow
import React from 'react'

import {
    StyleSheet,
    Text,
    View,
} from 'react-native'

export default function LoadingView({text='Loadingâ€¦'}: {text?: string}) {
  return (
    <View style={styles.container}>
      <Text>
        {text}
      </Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ffffff',
  },
})
