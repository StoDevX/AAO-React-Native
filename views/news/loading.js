import React from 'react'

import {
    StyleSheet,
    Text,
    View,
} from 'react-native'

export default class LoadingView extends React.Component {

    render() {
        return (
            <View style={styles.container}>
                <Text>
                    Loading...
                </Text>
            </View>
        )
    }
}

var styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor : '#ffffff',
    },
});
