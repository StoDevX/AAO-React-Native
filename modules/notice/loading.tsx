import * as React from 'react'

import {NoticeView} from './notice'

export function LoadingView({text = 'Loadingâ€¦'}: {text?: string}): JSX.Element {
	return <NoticeView spinner={true} text={text} />
}
