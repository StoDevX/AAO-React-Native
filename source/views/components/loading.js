// @flow
import React from 'react'
import {NoticeView} from './notice'

export default function LoadingView({text='Loadingâ€¦'}: {text?: string}) {
  return (
    <NoticeView text={text} spinner={true} />
  )
}
