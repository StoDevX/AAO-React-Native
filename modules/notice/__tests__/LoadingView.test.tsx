import React from 'react'
import {describe, expect, test} from '@jest/globals'

import TestRenderer from 'react-test-renderer'
import {LoadingView} from '../loading'

describe('LoadingView', () => {
	test('it displays "Loadingâ€¦" when no text is supplied', () => {
		const tree = TestRenderer.create(<LoadingView />).toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('it displays text when text is supplied', () => {
		const tree = TestRenderer.create(<LoadingView text="foo bar" />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
