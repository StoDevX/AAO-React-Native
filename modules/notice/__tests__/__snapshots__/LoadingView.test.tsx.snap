// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingView it displays "Loading…" when no text is supplied 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Loading…
  </Text>
</View>
`;

exports[`LoadingView it displays text when text is supplied 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": {
          "semantic": [
            "systemBackground",
          ],
        },
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
  <Text
    selectable={true}
    style={
      [
        {
          "color": {
            "semantic": [
              "label",
            ],
          },
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    foo bar
  </Text>
</View>
`;
